subsys/bluetooth/mesh/light_ctrl_regulator.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <bluetooth/mesh/light_ctrl_regulator.h>
+#include <bluetooth/mesh/light_ctrl_srv.h>
+
+#define REG_INT CONFIG_BT_MESH_LIGHT_CTRL_SRV_REG_INTERVAL
+
+static void reg_step(struct k_work *work)
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct bt_mesh_light_ctrl_spec_regulator *spec_reg = CONTAINER_OF(
+		dwork, struct bt_mesh_light_ctrl_spec_regulator, timer);
```

Comment:
```suggestion
	struct bt_mesh_light_ctrl_spec_regulator *spec_reg = CONTAINER_OF(
		work, struct bt_mesh_light_ctrl_spec_regulator, timer.work);
```
A matter of preference maybe, but a little more compact IMO

Reply:
I see that this is just a copy/paste of the original implementation, and that this way of doing it was done as a separate PR by Pavel. Might be a good reason for doing it in this manner.